pallets/parachain-staking/src/migrations.rs

```diff
@@ -12,35 +12,69 @@ use frame_support::{
 	weights::Weight,
 };
 
-const CURRENT_STORAGE_VERSION: StorageVersion = StorageVersion::new(7);
-const TARGET_STORAGE_VERSION: StorageVersion = StorageVersion::new(8);
+#[derive(Default)]
+enum Versions {
+	V7 = 7,
+	V8 = 8,
+	#[default]
+	V9 = 9,
+}
 
 pub(crate) fn on_runtime_upgrade<T: Config>() -> Weight {
 	upgrade::Migrate::<T>::on_runtime_upgrade()
 }
 
 mod upgrade {
 	use super::*;
+	use crate::pallet::{OLD_STAKING_ID, STAKING_ID};
+	use frame_support::traits::{LockableCurrency, WithdrawReasons};
+	use pallet_balances::Locks;
 
 	#[storage_alias]
 	type RewardRateConfig<T: Config> = StorageValue<Pallet<T>, RewardRateInfo, ValueQuery>;
 
-	/// Migration implementation that renames storage HardCap into MaxCurrencySupply
+	/// Migration implementation that deletes the old reward rate config and changes the staking ID.
 	pub struct Migrate<T>(sp_std::marker::PhantomData<T>);
 
 	impl<T: Config> Migrate<T> {
 		pub fn on_runtime_upgrade() -> Weight {
 			let mut weight_writes = 0;
-			let weight_reads = 0;
+			let mut weight_reads = 0;
 			let onchain_storage_version = Pallet::<T>::on_chain_storage_version();
-			if onchain_storage_version.eq(&CURRENT_STORAGE_VERSION) {
-				TARGET_STORAGE_VERSION.put::<Pallet<T>>();
-				log::error!("Migrating parchain_staking to V8");
+			if onchain_storage_version < StorageVersion::new(Versions::default() as u16) {
```

Comment:
Your implementation logic is correct. However, I want to ask why we want to keep the migration logic from V7 to V8? Could we deal with V8 to V9 only?

Reply:
I'm not sure i get it. you want to remove the migration from v7 to v8? then people on V7 will miss some of the migration no ?